core/src/main/scala/org/bitcoins/core/crypto/Sign.scala

```diff
@@ -58,3 +59,17 @@ object Sign {
     }, publicKey)
   }
 }
+
+/** A signing interface for [[ExtKey]] */
+trait ExtSign extends Sign { this: ExtPrivateKey =>
```

Comment:
I don't think it is good to require that all `ExtSign`s be `ExtPrivateKey`s and that instead you should make the `def sign` into a `def deriveAndSignFunction: (ByteVector, BIP32Path) => Future[ECDigitalSignature]` and implement this function on `ExtPrivateKey`. This way things will be consistent with the `Sign` trait.

Reply:
I also think that we need consistency between the `ExtPrivateKey - ECPrivateKey` and the `ExtSign - Sign` relationships. Either both `ExtPrivateKey` and `ExtSign` has-a `ECPrivateKey` and `Sign`, or both is-a. Currently we have `ExtSign is-a Sign` and `ExtPrivateKey has-a ECPrivateKey`